packages/react/src/components/ContentSwitcher/ContentSwitcher.js

```diff
@@ -54,6 +62,12 @@ export default class ContentSwitcher extends React.Component {
   }
 
   getChildren(children) {
+    const switchRefs =
```

Comment:
Would we want to mirror the structure from UI Shell where we pass in a reference to a function for the ref: https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/UIShell/HeaderNavigation.js#L79-L87 ?

This way we don't have to iterate through the children twice and can render in one pass

Reply:
This optimization was part of @asudoh's suggested changes -- could you elaborate on why we would want to mirror the UI shell code here? I don't have much experience working on UI shell so not much context.

Reply:
One thing that will come up here is that we are passing an anonymous function as a `ref` which will cause it to be called twice, setting it up with a named function should eliminate this case 👍